### ISSUE TITLE
Runs out of memory loading PLATEAU 3D Tiles 

### ISSUE THREAD

**Issue by joseph-kaile** [CesiumGS/cesium-unreal#746]
Loading Shinjuku Ward causes crash after running out of memory. The no-texture version does not crash.
![image](https://user-images.githubusercontent.com/97980867/152232510-fa5340f9-f553-479f-9c71-a9e60ca7b05c.png)

**Comment by joseph-kaile** [comment]
Yes, sorry for the late reply: https://www.geospatial.jp/ckan/dataset/plateau-tokyo23ku-3dtiles-2020

**Comment by azrogers** [comment]
I can confirm this is still an issue. The performance of this tileset in Cesium for Unreal is unbelievably bad. Starting at the mark here, the first bump is just from adding the tileset - nothing even rendering on screen yet. The second bump is when I actually navigated to the location of the tileset (around -3954807.0016528545, 3353151.5122543136, 3701405.140681766 in ECEF coordinates). GPU texture memory almost immediately ran out, Unreal and the rest of my computer froze, and the memroy kept climbing until around 48gb when I was finally able to kill the process. In another previous trial run without a profiler attached, it indeed crashed from running out of memory.

![Image](https://github.com/user-attachments/assets/0e9fe250-ed13-40db-bd26-a775c59f0602)

**Comment by azrogers** [comment]
It also seems to have caused an issue I've never seen before and never really even considered could be possible - completely corrupting everything on one of my monitors. My best guess is that it crashed the driver and somehow reverted the display to the worst available color mode, since it can't be screenshotted and everything on that monitor looks fine if I drag it to the other. Bizarre. I just had to take a picture to prove I wasn't going crazy:
![Image](https://github.com/user-attachments/assets/f446c55d-27e9-4f17-9ba2-e9a1236334c9)

**Comment by azrogers** [comment]
After analyzing the `b3dm` files with 3d-tiles-tools, I think I've found the culprit for this poor performance: the sheer number of images used in this tileset. Just `data0.b3dm` has 485 images spread across 486 primitives - yes, basically one texture per primitive. `data8.b3dm` has 302 images, and the largest b3dm files have similar numbers. I don't know that there's a way that we can optimize this on the runtime side - I think it needs to be optimized on the tiling side.

**Comment by kring** [comment]
@azrogers does this tileset working equally poorly in CesiumJS then?

**Comment by azrogers** [comment]
@kring I wouldn't necessarily say *equally* poorly. It's similarly unusable, slowing rendering to a handful of frames per second - if that - and accumulating more and more memory usage. But it doesn't seem to fill up VRAM, fill up regular RAM, or crash the graphics driver like it does in Unreal. But I would chalk this up to browser sandboxing. Regardless, it doesn't work well in either.

**Comment by kring** [comment]
Ok thanks @azrogers. One more quick question for you: you said that these hundreds of images are _embedded_ in the glTFs, right? Not referenced externally by URL.

**Comment by azrogers** [comment]
@kring Yep. Here's a sample of one of the b3dms, converted to glTF: https://gist.github.com/azrogers/a8140ccd21660a5264979c040edd534b
